### PR TITLE
  [DS-3656]  DOIOrganiser CLI does not change the database anymore

### PR DESCRIPTION
**Pull request by ssolim**
HibernateDBConnection add session.flush in uncacheEntity 6_x

### CODE REVIEW

**Comment by mwoodiupui** [comment]
Just a note:  we may be coming back to this code later for performance reasons.  uncacheEntity is mildly recursive and can (now) result in multiple flushes (up to eight, for a Collection).  If this is a problem, then probably the logic needs to move to an internalUncacheEntity() that does no flushing, wrapped in uncacheEntity which just delegates to internalUncacheEntity() and then does one flush.  All the recursion would be confined to internalUncacheEntity().  Likely it is best to treat this as premature optimization, for now, but I noticed it and wanted to capture the thought.

File: dspace-api/src/main/java/org/dspace/core/HibernateDBConnection.java [hunk 1]
```diff
@@ -240,7 +240,10 @@ private void configureDatabaseMode() throws SQLException {
                 }
             }
 
+            if(!readOnlyEnabled && getSession().isDirty()) {
+                getSession().flush();
```
Comment (mwoodiupui) [inline dspace-api/src/main/java/org/dspace/core/HibernateDBConnection.java]: Do we want getSession().clear() here?

**Comment by tomdesair** [comment]
This is (I think) prevented by the `getSession().isDirty()` check. This test will only return "true" is there are "unflushed" statements. Once you call `getSession().flush()` your session is "clean" again and it will not be executed again by the next recursive call. But it's best that we double check this by adding some debug/trace logging maybe.

**Comment by tdonohue** [comment]
On Slack #dev channel today, @ssolim recommended reverting the changes in this PR, and instead fixing this issue using newly added commit() messages like this: ssolim/DSpace@ac3ecbf

**Comment by ssolim** [comment]
I will close this as https://github.com/DSpace/DSpace/pull/1823/files is preferred solution

**Pull request closed without merging**